### PR TITLE
Add Devise

### PR DESCRIPTION
**Pull request by Retky**
### ☑️ [Requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md):

- ✔️ Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- ✔️ [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - ✔️ Can register a new user.
  - ✔️ User logs in with a combination of email and password.
  - ✔️ Hashed password should be stored in the database.
  - ✔️ Ask for confirmation of email.
  - ✔️ Can reset password.
- ✔️ [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+  # ==> Controller configuration
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
```
Comment (omar-muhamad) [inline config/initializers/devise.rb:28]: - [x] When I sign up for a new account I see the confirmation message which supposes to be sent to the registered email on the screen directly and this is not supposed to happen, So please when you click sign up make the confirmation message sent to the email you registered with.

[Hint]: See this [video ](https://youtu.be/sVJRkTxqlSQ?t=344)to know how to do it. (I send a link with the specific time talking about the mail sender but I suggest to see the whole video for better understanding)

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Retky and @sediqullahbadakhsh,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: config/initializers/devise.rb (hunk 1 above)
Comment (Retky) [reply config/initializers/devise.rb:28]: Oh, yea sorry I forgot to mention I'm using the `gem 'letter_opener'` the message is only for the develop preview as you can see in the gems file.
[Letter Oppener](https://github.com/ryanb/letter_opener#:~:text=Rails%20Setup%20First%20add%20the%20gem%20to%20your,in%20config%2Fenvironments%2Fdevelopment.rb%20config.action_mailer.delivery_method%20%3D%20%3Aletter_opener%20config.action_mailer.perform_deliveries%20%3D%20true)

**Review by Davidosky007 — APPROVED** [review]
Hi @Retky @sediqullahbadakhsh 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Well written readme file 👍 
* Good looking PR description 👍 
* Proper use of Gitflow 👍 
* Nice job using the letter_opener gem 👏 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**